app/__init__.py

```
@@ -100,12 +100,13 @@ def __getattribute__(self, name):
         _spec.submodule_search_locations = [os.path.dirname(__file__)]
-        __spec__ = _spec
-    else:
-        __spec__ = cast(ModuleSpec, None)
+    _module_spec: Optional[ModuleSpec] = _spec
 else:
     # Fallback if spec creation fails
-    __spec__ = cast(ModuleSpec, None)
+    _module_spec = cast(ModuleSpec, None)
+
+# Assign module __spec__ once via setattr to avoid rebinding builtins directly
+setattr(_this_module, "__spec__", _module_spec)
```

Ruff (0.13.2)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)